### ISSUE TITLE
Infoterminal Y4

### ISSUE THREAD

**Issue by denisehe** [strands-project/aaf_deployment#348]
@marc-hanheide @Jailander 
Hi there - can you please send us the nice picture you produced during the review - showing the infoterminal points and frequency of use? 

Thanks a lot :-)

so @capamat-aaf has a first glimps on where the robot is going to show this task

**Comment by capamat-aaf** [comment]
[info-screen-start.zip](https://github.com/strands-project/aaf_deployment/files/492987/info-screen-start.zip)
[Infoterminal Recommendations.docx](https://github.com/strands-project/aaf_deployment/files/493082/Infoterminal.Recommendations.docx)

@marc-hanheide @Jailander 
Hello, here are some drafts for the adapted infoscreen design, which I designed in Balasmiq Mockups and GIMP, so unfortunatley with a couple of limitations. 
I hope the general idea can be seen tough; I also added a doc where the changes/recommendations are writtten down. 
Waiting for you to tell me which aspects can be implemented. 

//The Balsamiq parts were created by the team before my arrival and the rest I can still polish, if need be.
The Balsamiq are just mockups, so I wasn't able to quickly adapt them to our needs (e.g. bigger spacing).

**Comment by marc-hanheide** [comment]
Consensus after meeting:
- suggested info screen accessible through big button (hiding the actual information)
- menu always visible
- _not_ displaying info-terminal while driving.

**Comment by capamat-aaf** [comment]
• for food menu have two buttons next to each other to select the desired one
• font size 14pt for mutliple lined text (should be around 18px, which looked bigger tan before, iirc)
• spacing between mutliple lined text (min))1.5-Line or Double Spacing, in any way so that scrolling can be avoided. I don't know what parameters are needed so I don't know how to convert it; here's a pic:

![image](https://cloud.githubusercontent.com/assets/21201058/19273416/bdc63bd8-8fcc-11e6-966d-5665aba234ac.png)

**Comment by capamat-aaf** [comment]
@marc-hanheide some observations regarding the info screen from today's excessive testing session :)
• Could you increase the Line Spacing for the text in News?
• Could I ask you for the link to the resident menu? Their menu is a bit more complex and I'ld like to try out the layout
• the font size in the menu is bigger in FF compared to Chrome, haven't checked the other items, but so far that's not a problem
• bottom(?)bar shows up in both FF and Chrome, in every menu point not just the menu
![img_20161111_1158343](https://cloud.githubusercontent.com/assets/21201058/20214786/8bf680f2-a810-11e6-8916-dba9ba5bb17d.jpg)
• Once it finished an info-terminal task and started moving, it retained the last screen that we opened, e.g. we checked the menu and then it moved with the menu being displayed. 
I'm not sure what the "on-the-move" screen should be; is that the one with the clock and suggesting news/weather?

And on a side note, on the STRANS laptop I'm viewing the Info-Screen through 192.168.20:8080/news, for example, and here News and Photos are different from what I see on Henry; I assume I'm using the wrong link. But this way I can't see what those things look like.

**Comment by marc-hanheide** [comment]
@capamat-aaf 

> @marc-hanheide some observations regarding the info screen from today's excessive testing session :)
> • Could you increase the Line Spacing for the text in News?

Yes, done in 111183c?

> • Could I ask you for the link to the resident menu? Their menu is a bit more complex and I'ld like to try out the layout

https://docs.google.com/document/d/1q4Aw0bPPeO9joDFGVbb59davNwdYdaSIWq1scbTkL3g/edit?usp=sharing

> • the font size in the menu is bigger in FF compared to Chrome, haven't checked the other items, but so far that's not a problem

OK, strange.

> • bottom(?)bar shows up in both FF and Chrome, in every menu point not just the menu
> ![img_20161111_1158343](https://cloud.githubusercontent.com/assets/21201058/20214786/8bf680f2-a810-11e6-8916-dba9ba5bb17d.jpg)

yes, I don't know how to avoid that :-(

> • Once it finished an info-terminal task and started moving, it retained the last screen that we opened, e.g. we checked the menu and then it moved with the menu being displayed. 
> I'm not sure what the "on-the-move" screen should be; is that the one with the clock and suggesting news/weather?

Yes, that's still on the list of things to do. Plan was to just have the time and some very basic information on it.

> And on a side note, on the STRANS laptop I'm viewing the Info-Screen through 192.168.20:8080/news, for example, and here News and Photos are different from what I see on Henry; I assume I'm using the wrong link. But this way I can't see what those things look like.

192.168.20:8080/news was only for test purposes. I'm surprised it is still running. It shouldn't. Ignore it.

**Comment by capamat-aaf** [comment]
Brilliant, thank you!
Can't view the line spacing changes, wouldn't know where now :P 
So basically, the link that I need to view the info terminal is the same address that Henry uses; but I guess I'll only be able to do that once it's running on him?

**Comment by capamat-aaf** [comment]
@marc-hanheide 
Sorry, about line spacing again; it's not top priority right now, but for the record: 
Somehow it still doesn't look good.
Here's a screenshot: apart from horrible news being displayed, there is plenty of room below. 
Perhaps you could increase it to double spacing? Or, did the 1.5 apply? It looks so narrow to me...
![line-spacing](https://cloud.githubusercontent.com/assets/21201058/20525496/043aab80-b0c0-11e6-9ed0-caae82d3d6c3.JPG)

**Comment by marc-hanheide** [comment]
Odd, I'm sure I fixed it. This could be due to the outdated @gestom workspace we have to use at the moment. Let's wait and say until that is fixed.

**Comment by capamat-aaf** [comment]
Another non-urgent question: is there a link that I can access through the STRANDS laptop/network where I can view the info terminal screens? The "testing url" doesn't work anymore.
The menus in google docs have a different layout compared to the info terminal and since they get cut off if they're too long, I was able to preview them on the laptop.

**Comment by capamat-aaf** [comment]
Minor thing: the "hover" function is showing the links eg localhost:8090/menures
But it seems the workaround might be tedious
(http://stackoverflow.com/questions/5605481/css-html-disable-link-hover-text)

**Comment by cdondrup** [comment]
Is that a problem? It does that in all the interface I guess. In the walking group as well.
But people can't do anything with that information so I don't think it needs to be fixed if that means replacing every link with a javascript call.

Also, it shows this sometimes even when an external component changes the webpage... So if my program triggers a new webpage to be shown, it still shows the hover thingy. I wouldn't even know where to stop that from happening.

**Comment by capamat-aaf** [comment]
It's only a "problem" when they keep the mouse at the menu bar and looks ugly.
But indeed, there are no functionality issues
![img_20161125_1149389](https://cloud.githubusercontent.com/assets/21201058/20623030/e56646ae-b305-11e6-89a2-6e6ba7893a60.jpg)

**Comment by capamat-aaf** [comment]
Wow, this is really bizarre
![img_20161130_0828117](https://cloud.githubusercontent.com/assets/21201058/20744116/42d0cf26-b6da-11e6-858b-819233691ccb.jpg)

@marc-hanheide it looks fine in google docs, and so far it was never an urgent issue (I could see it in the control UI if people clicked on it and then correct it)
But if it's like that it'd be inconvenient to go to the robot every time to check if the layout looks fine.
Can I preview/see it "on" Henry somehow?

**Comment by marc-hanheide** [comment]
OK @capamat-aaf, I now launch a test GUI automatically on the control PC. You can go to http://192.168.200.20:9999 to see it. Note: The facebook news are not displayed as the OAuth token isn't available on this one.

**Comment by capamat-aaf** [comment]
Thank you very much, it is tough to always find the best size of items and this facilitates my job quite a bit

**Comment by capamat-aaf** [comment]
@marc-hanheide 
Is there a way to decide the order of pictures in the gallery of the info-terminal?
Perhaps naming them alphabetically?

**Comment by marc-hanheide** [comment]
that's beyond my knowledge... @cdondrup do you recall?

**Comment by cdondrup** [comment]
Sorry, no idea. This whole media server thing was always black magic to my eyes...

**Comment by capamat-aaf** [comment]
Ok, I'll try things out
Edit: so apparently it's like this: in the file manager they are sorted alphabetically.
In the sets, it's entropy for the ones in set or not (also when adding them), but regardless of that; the one that gets added first will be displayed on the robot first, etc, which is very good.

**Comment by capamat-aaf** [comment]
@marc-hanheide 
Some minor issues/questions/"Schönheitsfehler":
• the emergency stop overlaps with "Übermorgen" on weather. Could the content be moved down a bit? There is plenty of space until the menu bar
• and I finally found a (n incomplete) bellbot screenshot
the "address" field could be the bigger one and on top in my opinion, instead of delete entry 
(I think the upper one is what it looks like now, not 100% certain)

EDIT:  nevermind, I just figured why it is the way it is. The dop-down suggestion list needs space to go down and shouldn't overlap the keyboard. Sorry, took me a while.
